InventoryIndexer/Model/ResourceModel/UpdateLegacyStockStatus.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryIndexer\Model\ResourceModel;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\InventoryCatalogApi\Model\GetProductIdsBySkusInterface;
+
+/**
+ * Update legacy stock status for given skus.
+ */
+class UpdateLegacyStockStatus
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resource;
+
+    /**
+     * @var GetProductIdsBySkusInterface
+     */
+    private $getProductIdsBySkus;
+
+    /**
+     * @param ResourceConnection $resource
+     * @param GetProductIdsBySkusInterface $getProductIdsBySkus
+     */
+    public function __construct(
+        ResourceConnection $resource,
+        GetProductIdsBySkusInterface $getProductIdsBySkus
+    ) {
+        $this->resource = $resource;
+        $this->getProductIdsBySkus = $getProductIdsBySkus;
+    }
+
+    /**
+     * Update legacy stock status for given skus.
+     *
+     * @param array $dataForUpdate
+     * @param string $connectionName
+     */
+    public function execute(array $dataForUpdate, string $connectionName): void
```

Comment:
Does it make sense to pass the connection name here? Is there any case when a connection can be different than default?